### PR TITLE
News endpoint

### LINKED ISSUE #3: Add "news" entity to backend with GET and POST route
- Add migration to add "news" entity to database with fields title and description
- Add GET route that returns all news
- Add POST route that creates a new news row in the database

ADD POSTMAN SCREENSHOT SHOWING ROUTES WORKING

### PR DESCRIPTION
**Pull request by surbhi-gulati**
Closes #3.

+ Introduced a News entity to the database, with fields: id (int), title (string), content(string), createdAt (datetime)
+ Created GET and POST endpoint in /pages/api/news
+ Validated POST type of news items
  + Created 'createNewsRequest.dto.ts' - defines news schema and infers the news type
  + Added 'BadRequestError' class in utils to facilitate checking type by parsing via Zod
  + Revise makeRoute util to transform internal server errors to 400 errors if instance of BadRequestError
+ Wrote end-to-end tests for GET / POST and malformed POST requests using Jest
POST_news_endpoint_success
GET_news_endpoint_success

### CODE REVIEW

**Comment by rymaju** [comment]
This is an *excellent* PR description 💯

File: test/api/news.test.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import { getNewsHandler, postNewsHandler} from '../../pages/api/news';
+import httpMocks, { Body } from 'node-mocks-http';
+import prisma from '../../prisma/prisma';
+import { News } from '@prisma/client';
+import { NewsWrite } from '../../lib/api/createNewsRequest.dto';
+import { BadRequestError } from '../../lib/utils/errors/badRequestError';
+
+const createNews0:NewsWrite = {
```
Comment (rymaju) [inline test/api/news.test.ts:8]: 📌 We might want these in an examples or fixtures folder

**Review by rymaju — APPROVED** [review]
LGTM

File: pages/api/news.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import { BadRequestError } from '../../lib/utils/errors/badRequestError';
+
+export const getNewsHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  res.status(200).json({ news: await prisma.news.findMany() });
+};
+
+export const postNewsHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const validatedNews = validateCreateNewsRequest(req.body);
+  await prisma.news.create({ data: validatedNews });
+  res.status(201).send('Success!');
```
[the remark this answers is not in the document]
Comment (jessicasu77) [reply pages/api/news.ts:15]: What is the consistent POST behavior across the app- I think this is the first POST endpoint?

File: test/api/news.test.ts (hunk 1 above)
Comment (jessicasu77) [reply test/api/news.test.ts:8]: @varunthakkar1 would we want an examples folder- where in the file directory would be best?

Comment (vthakkar1) [reply test/api/news.test.ts:8]: @jessicasu77 test/api/data/newsFixtures.ts maybe? or maybe test/api/fixtures, what do you think @rymaju

**Review by varunthakkar1 — APPROVED** [review]
Really really great work!!! Make sure to address Ryan's comments before merging

File: test/api/news.test.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+const news0:News = {
+  id: 0,
+  title: 'First news',
+  content: ':)',
+  createdAt: new Date('1997-08-17')
+};
+
+const news1 = {
```
Comment (varunthakkar1) [inline test/api/news.test.ts:24]: We should make this type News, aka `const news1: News = {`

Comment (varunthakkar1) [reply test/api/news.test.ts:24]: We should also test that having an empty string for title or content is invalid. We should also test that having no title is invalid

File: pages/api/news.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (rymaju) [reply pages/api/news.ts:15]: To clarify, I don't think that we would want to respond with `Success!` for every 201, its not wrong but it means that we would always need to say `Success!`, not really something I would like to enforce everywhere.

[the remark this answers is not in the document]
Comment (jessicasu77) [reply pages/api/news.ts:15]: Gotcha, but is not sending any response problematic? Referring to when Surbhi and I were testing the POST on postman and thought it wasn't succeeding because it timed out due to not sending any response (we realized it worked by going to the endpoint in the browser and seeing what was possted)?

[the remark this answers is not in the document]
Comment (rymaju) [reply pages/api/news.ts:15]: I thought `.status` was enough, you might also need `.end` though. i.e.
`res.status(201).end();`

File: test/api/news.test.ts (hunk 1 above)
Comment (jessicasu77) [reply test/api/news.test.ts:8]: @varunthakkar1  yup we put it in /test/api/examples/newsMockData.ts

File: pages/api/news.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jessicasu77) [reply pages/api/news.ts:15]: @rymaju yup we did it!

File: pages/api/news.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+export default makeRoute().get(getNewsHandler).post(postNewsHandler);
+
```
[the remark this answers is not in the document]
Comment (jessicasu77) [reply pages/api/news.ts:19]: Moved all the logic into /lib/api/newsLogic.ts
